jack-core/src/com/rapleaf/jack/transaction/DbPoolManager.java

```diff
@@ -82,6 +85,7 @@
   @Override
   public DB getConnection() {
     try {
+      metrics.update(true, connectionPool);
       return connectionPool.borrowObject();
```

Comment:
Should be `update` method be called after the `borrowObject` returns? If there is no idle connection, the `borrowObject` will be blocked and it can take some time. Calling `update` before it will under calculate the waiting time.